digdag-client/src/main/java/io/digdag/client/DigdagClient.java

```diff
@@ -427,17 +427,18 @@ public RestProject putProjectRevision(String projName, String revision, File bod
     @FunctionalInterface
     interface RequestWithFollowingRedirect<T>
     {
-        T invoke(WebTarget webTarget);
+        T invoke(WebTarget webTarget, Optional<WebTarget> previousWetTarget);
```

Comment:
This is minor thing but how about using `Optional<Response> redirectResponse` instead? `Response` provides more information to the caller because previousWebTarget is anyways available.